### PR TITLE
sensors: Blocking read with new API

### PR DESCRIPTION
**Pull request by teburd**
Rename the current sensor_read to sensor_read_async_mempool and create a new blocking sensor_read helper.

Depends on #71093

### CODE REVIEW

File: include/zephyr/drivers/sensor.h [hunk 1]
```diff
@@ -1005,7 +1005,50 @@ static inline int sensor_stream(struct rtio_iodev *iodev, struct rtio *ctx, void
...
+ * will wait for the read to complete before returning to the caller.
+ *
+ * @param[in] iodev The iodev created by @ref SENSOR_DT_READ_IODEV
+ * @param[in] ctx The RTIO context to service the read
+ * @param[in] buf Pointer to memory to read sample data into
+ * @param[in] buf_len Size in bytes of the given memory that are valid to read into
+ * @return 0 on success
+ * @return < 0 on error
+ */
+static inline int sensor_read(struct rtio_iodev *iodev, struct rtio *ctx, uint8_t *buf,
```
Comment (laurenmurphyx64) [inline include/zephyr/drivers/sensor.h:1021]: This is synchronous, yet it's in the #ifdef CONFIG_SENSOR_ASYNC_API - intentional?

**Review by laurenmurphyx64 — DISMISSED** [review]

File: tests/drivers/build_all/sensor/src/generic_test.c
Comment (ubieda) [inline tests/drivers/build_all/sensor/src/generic_test.c:1]: Would it make sense to add a test-case to also cover `sensor_read()`?

**Review by ubieda — DISMISSED** [review]
Looks good. It would be beneficial to validate `sensor_read` under a test-case, but it may as well be captured on a separate PR.

**Review by laurenmurphyx64 — DISMISSED** [review]

**Review by laurenmurphyx64 — APPROVED** [review]

**Review by ubieda — APPROVED** [review]

File: drivers/sensor/sensor_shell.c [hunk 2]
```diff
@@ -561,7 +561,7 @@ static int cmd_get_sensor(const struct shell *sh, size_t argc, char *argv[])
 
 	ctx.dev = dev;
 	ctx.sh = sh;
-	err = sensor_read(&iodev_sensor_shell_read, &sensor_read_rtio, &ctx);
+	err = sensor_read_async_mempool(&iodev_sensor_shell_read, &sensor_read_rtio, &ctx);
```
Comment (ubieda) [inline drivers/sensor/sensor_shell.c:564]: Do we still need to call `sensor_processing_with_callback()`?

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**